ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -240,11 +240,14 @@ private boolean deregisterHardDeleteMetric(String storeId) {
         MetricRegistry.name(PersistentIndex.class, prefix + "HardDeleteCaughtUp"));
   }
 
-  void initializeCompactorGauges(String storeId, final AtomicBoolean compactionInProgress) {
+  void initializeCompactorGauges(String storeId, final AtomicBoolean compactionInProgress, final boolean useDirectIO) {
     String prefix = storeId + SEPERATOR;
     Gauge<Long> compactionInProgressGauge = () -> compactionInProgress.get() ? 1L : 0L;
     registry.register(MetricRegistry.name(BlobStoreCompactor.class, prefix + "CompactionInProgress"),
         compactionInProgressGauge);
+    Gauge<Integer> compactionUseDirectIO = () -> useDirectIO == true ? 1 : 0;
```

Comment:
do we need a gauge for this? I would think a log message + the config value should suffice

Reply:
log is good. Fixed.